### ISSUE TITLE
[Request] Verification method to assure that code is formatted properly

### ISSUE THREAD

**Issue by Paden** [vvakame/typescript-formatter#15]
Something like:

```
$ tsfmt -verify <other options> <files>
```

The files would not be changed, only verified according to the specs.

**Comment by vyk2rr** [comment]
+1

**Comment by gpfarris** [comment]
+1

**Comment by cjlpowers** [comment]
+1

**Comment by dvalys** [comment]
+1

**Comment by poncho1984** [comment]
+1

**Comment by slavik925** [comment]
+1

**Comment by vvakame** [comment]
when do you want to use that option?

**Comment by Paden** [comment]
Formatting helps our team check-in code that looks as if it were written by one person, which helps readability.

Being able to verify that these files have been formatted correctly on tools like Travis CI or TeamCity would help our team make sure that code being checked in is in fact formatted.

**Comment by vvakame** [comment]
@Paden release typescript-fromatter 0.4.0. try it out.

**Comment by myitcv** [comment]
I'm seeing a problem with `--verify` in `v0.4.2` (I've substituted `\t` where appropriate below to remove any ambiguity):

```
$ cat js/services/banana.ts
export class Banana {
\tconstructor() {
\t\tconsole.log("Test");
\t}
}
$ tsc --version
message TS6029: Version 1.6.0-dev.20150803
$ tsfmt --version
0.4.2
$ tsfmt --verify js/services/banana.ts
js/services/banana.ts is not formatted
$ tsfmt js/services/banana.ts | diff js/services/banana.ts -
<no output>
$ tsfmt js/services/banana.ts | tsfmt --stdin --verify
temp.ts is not formatted
```

I'm using a fairly standard [`tslint.json`](https://gist.github.com/myitcv/d7a5996951be69624ba9)

Any ideas?

**Comment by vvakame** [comment]
@myitcv I can't reproduce your issue. I got below result.

```
$ tsfmt banana.ts | diff banana.ts -
--- banana.ts	2015-08-03 21:11:00.000000000 +0900
+++ -	2015-08-03 21:11:25.000000000 +0900
@@ -2,4 +2,5 @@
 	constructor() {
 		console.log("Test");
 	}
-}
\ No newline at end of file
+}
+
```

**Comment by myitcv** [comment]
Thanks for checking. Can I just ask, what platform are you working on? Linux/Mac/Windows?

**Comment by vvakame** [comment]
I'm using Mac OS X.

**Comment by myitcv** [comment]
That's my fault, because I didn't accurately copy my failing example. In any case I think I've worked out what's going on.

Let me try again, showing `input.ts`:

```
export class Banana {
    constructor() {
        console.log("Test");
    }
}

```

Notice the blank line on the end (GitHub shows it in a strange way). Checking via `hexdump`:

```
$ hexdump -C input.ts
00000000  65 78 70 6f 72 74 20 63  6c 61 73 73 20 42 61 6e  |export class Ban|
00000010  61 6e 61 20 7b 0a 20 20  20 20 63 6f 6e 73 74 72  |ana {.    constr|
00000020  75 63 74 6f 72 28 29 20  7b 0a 20 20 20 20 20 20  |uctor() {.      |
00000030  20 20 63 6f 6e 73 6f 6c  65 2e 6c 6f 67 28 22 54  |  console.log("T|
00000040  65 73 74 22 29 3b 0a 20  20 20 20 7d 0a 7d 0a 0a  |est");.    }.}..|
```

This means that `input.ts` ends in *two* line feeds.

[This code](https://github.com/vvakame/typescript-formatter/blob/3843e40fd97a7b7ed10235e3a9c1a9bcd15ea67c/lib/index.ts#L105-L108) then gobbles both line feeds, and adds back only one. If the intention here is to ensure we only have one line feed at the end of the file, I'm good with that.

However, the problem arises when used in the mode `tsfmt input.ts` (i.e. where the formatted output is printed to `stdout`). In this situation *another* line feed is being added by [`console.log`](https://github.com/vvakame/typescript-formatter/blob/3843e40fd97a7b7ed10235e3a9c1a9bcd15ea67c/lib/cli.ts#L112)

_(Indeed this is confirmed by the fact that if you run `tsfmt -r input.ts` the result does check correctly)_

So I think all of our problems here are solved by replacing `console.log` with a `process.stdout.write` call (plus a couple of tweaks to add `\n` to messages)

Thoughts?

**Comment by myitcv** [comment]
Worth noting that this `console.log` problem will cause issues for any editor that uses a `tsfmt --stdin` approach (then reading the results from `stdout`)

**Comment by myitcv** [comment]
I've just raised #19 to flesh out a fix for this
